### DISCUSSION TITLE
help, i have executed a strategy to buy and sell basis the pivot points which are already the part of the data being run through however, no trades are being taken by the following code

### DISCUSSION THREAD

**Discussion by bafnarahul2** [kernc/backtesting.py#762] [General]
class PdhR1(Strategy):
    
    
    def init(self):
       print(self.data)
       
       self.PDH = self.data.PDH
       
       self.PDL = self.data.PDH
       self.TC= self.data.index
       self.R1 = self.data.r1
       self.S1 = self.data.s1
       #print(self.PDH,self.PDL,self.TC,self.R1,self.S1) 
   
    def next(self):
       price = self.data.Close[-1]
       print(price)
       # print(self.PDL)
       if self.TC.time[0]>=dt.time(15,15):
           self.position.close()
       elif price>self.PDH[0] and price>self.R1[0] and not self.position:
            Self.buy(size=1)
       elif price<self.PDL[0] and price<self.S1[0] and not self.position:
            Self.sell(size=1)

**Comment by kernc** [comment]
Note, [`Strategy.data`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.data) arrays are ordered in time-ascending order. Therefore, use:
```py
if self.data.index[-1] > ...:
```

**Reply by bafnarahul2** [reply]
I did what you suggested but still i get only one trade and no exit or reentry. below is the entire strategy class code for your reference


class PdhR1(Strategy):
    
    def init(self):
       #print(self.data)
       
       self.PDH = self.data.PDH
       
       self.PDL = self.data.PDL
       
       self.TC= self.data.index
       
       self.td= self.data.td
       
       self.R1 = self.data.r1
       self.S1 = self.data.s1
       
# to calculate day based values  
    def next(self):
       price = self.data.Close[-1]
       timet=self.data.index.time[-1]
       
       
       
          
       if self.data.index.time[-1]>=dt.time(9,20):
           if (price>=self.PDH and price>self.R1):
         
              self.buy()
           elif (price<self.PDL and price<self.S1):
        
              self.sell()
       elif self.data.index.time[-1]>=dt.time(15,15) :
         
           if self.position.is_long:
              self.position.close()
           elif self.position.is_short:
              self.position.close()         
      
            
       
    
bt=Backtest(Y,PdhR1,cash=100000000)

stats=bt.run()
